### ISSUE TITLE
Bug with cache when aggregate

### ISSUE THREAD

**Issue by antoine-conserto** [turbot/steampipe#4378]
Hi,

**Describe the bug**
When using cache on a table that calls an api with field selection and filter, then an aggregation like sum or count disctint, sometimes there is a data inconsistency.

**Steampipe version (`steampipe -v`)**
Steampipe v0.23.5

**To reproduce**

Use this sample with chaos plugin and table `chaos_bug_cache_sum` : https://github.com/antoine-conserto/steampipe-plugin-chaos/tree/feature/bug-cache-sum

Run steampipe service
Run the query : 
`SELECT SUM(c.amount) AS "sum", COUNT(distinct id) AS "count"
FROM chaos.chaos_bug_cache_sum c `

Same problem with steampipe query directly.

**Expected behavior**

This query should always return 200,000 in sum and 10,000 in distinct count.

![expected](https://github.com/user-attachments/assets/73722a79-17d2-4f4c-85d9-2e8ccd3ced93)
[database-2024-09-05_ok.log](https://github.com/user-attachments/files/16893580/database-2024-09-05_ok.log)
[plugin-2024-09-05_ok.log](https://github.com/user-attachments/files/16893581/plugin-2024-09-05_ok.log)

**Additional context**

**It is not systematic. Sometimes I have the right result, if this is the case you have to restart the service or clear cache**

Sometimes I only got the sum wrong but the distinct count was good. 

I don't have the problem if I disable the cache on the table.

It's only the first query when caching that seems problematic. If I rerun after caching, I get the correct result and don't reproduce.

Exemple KO 1
![ko_1](https://github.com/user-attachments/assets/1b07b0fd-2ce2-4c44-8f3b-0dac0adfec2c)
[database-2024-09-05_ko_1.log](https://github.com/user-attachments/files/16893604/database-2024-09-05_ko_1.log)
[plugin-2024-09-05_ko_1.log](https://github.com/user-attachments/files/16893606/plugin-2024-09-05_ko_1.log)

Exemple KO 2
![ko_2](https://github.com/user-attachments/assets/92c1c237-405e-4711-be91-426f5e1c5c94)
[database-2024-09-05_ko_2.log](https://github.com/user-attachments/files/16893605/database-2024-09-05_ko_2.log)
[plugin-2024-09-05_ko_2.log](https://github.com/user-attachments/files/16893607/plugin-2024-09-05_ko_2.log)

**Comment by NavSethi2006** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler

**Comment by pskrbasu** [comment]
@antoine-conserto Thanks for raising this detailed issue. I'll take a look into it and try to reproduce the issue.

**Comment by antoine-conserto** [comment]
Any update ?

**Comment by vbatychko-modeln** [comment]
Can confirm that with v0.24.2 and AWS plugin 0.147.0. This select does not give consistent result. Sometimes, 0, sometimes 2 or more.
Disabling cache fixes bug. Note that I'm using aggregation for multiple AWS accounts.

```
steampipe=> select count(*) from (select count(*), subnet_arn from aws_vpc_subnet group by subnet_arn having count(*)>1) t;
 count
-------
     2
(1 row)
```

**Comment by pskrbasu** [comment]
@antoine-conserto @vbatychko-modeln Apologies for the delay. I'll try this out today and update this thread.

**Comment by pskrbasu** [comment]
@antoine-conserto A small update on this. I was unable to reproduce the bug using your chaos table. I'm always getting the expected (This query should always return 200,000 in sum and 10,000 in distinct count.). I'll keep digging.

Meanwhile, I'll also try using the AWS query @vbatychko-modeln provided.

**Comment by antoine-conserto** [comment]
I also retested. 

In steampipe v0.24.2 
![Image](https://github.com/user-attachments/assets/4cc22c20-912a-4e19-8299-85d644a5cf42)


And with v1.0.0
![Image](https://github.com/user-attachments/assets/f5eeadfe-0d49-4d7e-98fd-55a7fb5bd974)


I did not change the sdk version (v5.10.4) on the chaos plugin. Maybe your refactor on the FDW fixed the problem, in any case it no longer seems relevant to me.

Thanks for watching @pskrbasu
